Runtime/Interaction/HPUIInteractorRayAngle.cs

```diff
@@ -7,69 +7,64 @@ namespace ubco.ovilab.HPUI.Interaction
     [Serializable, BurstCompile]
-    public struct HPUIInteractorRayAngle
+    public class HPUIInteractorRayAngle
     {
         [SerializeField] private float x;
         [SerializeField] private float z;
         [SerializeField] private float raySelectionThreshold;
-
+        private Vector3 leftRayDirection;
+        private Vector3 rightRayDirection;
+        [SerializeField] private bool isCached;
         public float X { get => x; }
         public float Z { get => z; }
         public float RaySelectionThreshold { get => raySelectionThreshold; set => raySelectionThreshold = value; }
-
         public HPUIInteractorRayAngle(float x, float z, float angleThreshold)
         {
+            Debug.Log("Creating New HPUIInteractorRayAngle");
             this.x = x;
             this.z = z;
             this.raySelectionThreshold = angleThreshold;
+            isCached = false;
+            this.rightRayDirection = GetDirection(x, z, false);
+            this.leftRayDirection = GetDirection(x, z, true);
         }
 
         public HPUIInteractorRayAngle(float x, float z)
         {
+            Debug.Log("Creating New HPUIInteractorRayAngle");
             this.x = x;
             this.z = z;
+            isCached = false;
             this.raySelectionThreshold = 1f;
+            this.rightRayDirection = GetDirection(x, z, false);
+            this.leftRayDirection = GetDirection(x, z, true);
         }
 
         public bool WithinThreshold(float dist)
         {
             return dist < raySelectionThreshold;
         }
 
-        // /// <summary>
-        // /// Get a quaternion that would rotate x angles around its right axis and z angles around its forward axis.
-        // /// </summary>
-        // private static Quaternion GetRotation(float x, float z)
-        // {
-        //     float yDist = Mathf.Sqrt(1 / (1 + Mathf.Pow(Mathf.Tan(x * Mathf.Deg2Rad), 2) + Mathf.Pow(Mathf.Tan(z * Mathf.Deg2Rad), 2)));
-        //     if (Mathf.Abs(x) > 90 || Mathf.Abs(z) > 90)
-        //     {
-        //         yDist = -yDist;
-        //     }
-
-        //     float xDist = Mathf.Tan(x * Mathf.Deg2Rad) * yDist;
-        //     float zDist = Mathf.Tan(z * Mathf.Deg2Rad) * yDist;
-
-        //     Vector3 newUp = Vector3.up * yDist + Vector3.right * xDist + Vector3.forward * zDist;
-
-        //     // Don't care about the forward axis, as long the the up vector gets to the right postion
-        //     Vector3 newforward = Vector3.Cross(newUp, Quaternion.Euler(0, 0, 90) * newUp);
-        //     Vector3 newRight = Vector3.Cross(newforward, newUp);
-        //     Debug.Log($"{Vector3.Dot(newUp, Vector3.Cross(newRight, newforward))}");
-
-        //     return Quaternion.LookRotation(newUp, newforward);
-        //     // return Quaternion.LookRotation(Vector3.up, newUp);
-        // }
+
+        public Vector3 GetDirection(bool isLeftHand)
+        {
+            if (isCached) return isLeftHand ? leftRayDirection : rightRayDirection;
```

Comment:
maybe don't inline this, expand this, and try to maintain the styling of the rest of the code base.

Reply:
little unclear about this